pkg/driver/driver_openstack.go

```diff
@@ -160,6 +161,29 @@ func (d *OpenStackDriver) Create() (string, string, error) {
 
 	metrics.APIRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "nova"}).Inc()
 
+	var useConfigDrive *bool
+	if len(serverNetworks) > 0 {
```

Comment:
Does it make sent to loop over all serverNetworks?
```suggestion
	for _, serverNetwork := range serverNetworks {
```

Reply:
I've removed all the auto-detection logic due to this. We can't make such assumptions.